app/src/main/kotlin/org/fossify/phone/activities/MainActivity.kt

```diff
@@ -290,6 +294,51 @@ class MainActivity : SimpleActivity() {
             } catch (ignored: Exception) {
             }
         }
+        pushStarredShortcuts()
+    }
+
+    private fun pushStarredShortcuts() {
+        if (VERSION.SDK_INT >= VERSION_CODES.R) {
+            val starred = cachedContacts.filter { it.starred == 1 && it.phoneNumbers.isNotEmpty() }
+            this.config.favoritesContactsOrder.let { order ->
+                if (!config.isCustomOrderSelected) {
+                    starred.sorted()
+                } else {
+                    val orderList = Converters().jsonToStringList(order).withIndex().associate { it.value to it.index }
+                    starred.sortedBy { orderList[it.contactId.toString()] }
+                }
+            }.take(3).forEachIndexed { index, contact ->
+                var number = if (contact.phoneNumbers.size == 1) {
+                    contact.phoneNumbers[0].normalizedNumber
+                } else {
+                    contact.phoneNumbers.firstOrNull { it.isPrimary }?.normalizedNumber
+                }
+                if (number == null) {
+                    val radioItems = contact.phoneNumbers.mapIndexed { index, item ->
+                        RadioItem(index, item.normalizedNumber, item)
+                    }
+                    RadioGroupDialog(this, ArrayList(radioItems)) {
+                        number = (it as PhoneNumber).normalizedNumber
+                    }
+                }
```

Comment:
Radio dialog here is bad UX, and it doesn't even work. Let's fall back to the first number when there is no primary number. Multiple numbers will be handled when https://github.com/FossifyOrg/Phone/issues/102 is done.

```suggestion
            val phoneNumbers = contact.phoneNumbers
            var number = if (phoneNumbers.size == 1) {
                phoneNumbers[0].normalizedNumber
            } else {
                phoneNumbers.firstOrNull { it.isPrimary }?.normalizedNumber
                    ?: phoneNumbers.firstOrNull()?.normalizedNumber ?: return@forEachIndexed
            }

```